Aplib.Tests/Belief/ExposedQueueTests.cs

```diff
@@ -0,0 +1,231 @@
+using System;
+using System.Collections.Generic;
+
+namespace Aplib.Core.Tests.Belief;
+
+public class ExposedQueueTests
+{
+    [Fact]
+    public void ExposedQueue_WhenInitialized_ShouldHaveCorrectMaxCountAndCount()
+    {
+        // Arrange
+        ExposedQueue<int> queue = new(3);
+
+        // Act
+        int maxCount = queue.MaxCount;
+        int count = queue.Count;
+
+        // Assert
+        Assert.Equal(3, maxCount);
+        Assert.Equal(0, count);
+    }
+
+    [Fact]
+    public void ExposedQueue_WhenInitializedWithArray_ShouldHaveCorrectMaxCountAndCount()
+    {
+        // Arrange
+        ExposedQueue<int> queue = new([1, 2, 3]);
+
+        // Act
+        int maxCount = queue.MaxCount;
+        int count = queue.Count;
+
+        // Assert
+        Assert.Equal(3, maxCount);
+        Assert.Equal(3, count);
+    }
+
+    [Fact]
+    public void ExposedQueue_WhenInitializedWithArrayAndCount_ShouldThrowExceptions()
+    {
+        // Arrange
+        void CountExceedsLength() => _ = new ExposedQueue<int>([1, 2, 3], 4);
+        void NegativeCount() => _ = new ExposedQueue<int>([1, 2, 3], -1);
+
+        // Assert
+        Assert.Throws<ArgumentOutOfRangeException>(CountExceedsLength);
+        Assert.Throws<ArgumentOutOfRangeException>(NegativeCount);
```

Comment:
Please write two separate tests for this. Asserting multiple things in a test is a test smell, as it is difficult to quickly pinpoint what the point of failure is (was it a high count or a negative count that caused this test to fail?)

In general, I'd think multiple asserts per test are allowed if you need to assert multiple thing with the same setup. This method uses two act methods and thus do not comply.